pages/recipes/[id]/index.js

```diff
@@ -9,22 +9,39 @@ import {
   StyledDetailsItem,
   StyledItemsRow,
   EditDeleteDiv,
-} from "@/components/StyledDetailsPage.js"
+} from "@/components/StyledDetailsPage.js";
 
 import {
   StyledButtonNo,
   StyledButtonYes,
   StyledDeleteButton,
 } from "@/components/Modal/ModalStyle.js";
 import { StyledDetailsItemIngredientsUl } from "@/components/StyledDetailsPage.js";
-
+import FavoriteButton from "@/components/FavoriteButton";
 
 export default function DetailsPage() {
   const [showModal, setShowModal] = useState(false);
+
   const router = useRouter();
   const { isReady } = router;
   const { id } = router.query;
-  const { data: recipe, isLoading, error } = useSWR(`/api/recipes/${id}`);
+  const {
+    data: recipe,
+    isLoading,
+    error,
+    mutate,
+  } = useSWR(id ? `/api/recipes/${id}` : null);
+
+  async function handleToggleFavorite(newStatus) {
+    await fetch(`/api/recipes/${id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ isFavorite: newStatus }),
```

Comment:
I would be careful here: you´re sending an object to update to your API route, but it only includes the `isFavorite` property which is something you might not want.

Reply:
Your API handles it this way and it would override the whole recipe with just this property:

```
if (request.method === "PUT") {
      const recipeData = request.body;
      await Recipe.findByIdAndUpdate(id, recipeData);
      response.status(200).json({ status: "Entry updated!" });
}
```

Reply:
I saw the function `handleToggleFavorite` at the top of this PR already, maybe you can find some way to have it as one (externalised) function or a part of it.
It´s not a must but it would follow clean code principles (don´t repeat yourself 😉).

Reply:
right now we can not see any problem as our recipes will not be changed when changing the favorite-State... maybe you could give us a hint for what we should change in our code ;-)

thanks a lot for your help!!